tools/jit/gen_unboxing.py

```diff
@@ -0,0 +1,182 @@
+    target: Union[Literal[Target.DECLARATION], Literal[Target.DEFINITION]]
+
+    @method_with_native_function
+    def __call__(self, f: NativeFunction) -> str:
+
+        if self.target is Target.DECLARATION:
+            # Note [The ATen Codegen Unboxing API]
+            # Similar to the ATen Operators API, ATen Codegen Unboxing API lives in the at::unboxing namespace, and
+            # will be used by codegen unboxing wrappers (CodegenUnboxingWrappers.cpp).
+            # The Wrappers will be registered into torch::jit::OperatorRegistry using RegisterOperators API.
+            #
+            # Important characteristics about the Codegen Unboxing API:
+            # (1) It follows the OperatorRegistry API.
+            #     This is kind of necessary to avoid overhead.
+            #     For example: if it followed the C++ API, then all of the faithful C++ factory functions
+            #     would need to wrap their arguments into TensorOptions only to unwrap them again.
+            # (2) Under the hood it calls C++ API.
+            return f"""
+// aten::{f.func}
+TORCH_API void {f.func.name.unambiguous_name()}(Stack & stack);
+"""
+        else:
+            sig_group = CppSignatureGroup.from_native_function(
+                f, method=(Variant.method in f.variants)
+            )
+            sig = sig_group.most_faithful_signature()
+            # parse arguments into C++ code
+            binding_list, code_list = convert_arguments(f)
+
+            # for each C++ argument, generate the conversion code
+            code_connector = "\n\t"
+            arg_connector = ", "
+            # function call and push back to stack
+            prefix = "self_base." if sig.method else "at::"
+            translated_args = translate(binding_list, sig.arguments(), method=sig.method)
+            args_str = f"{arg_connector.join(e.expr for e in translated_args)}"
+            if len(f.func.returns) == 0:
+                ret_str = ""
+                push_str = ""
+            else:
+                ret_str = "auto result_ = "
+                push_str = """
+    pack(stack, std::move(result_));
+                """
+            return f"""
+// aten::{f.func}
+TORCH_API void {f.func.name.unambiguous_name()}(Stack & stack) {{
+    {code_connector.join(code_list)}
+
+    drop(stack, {len(binding_list)});
+
+    {ret_str}{prefix}{sig.name()}({args_str});
+    {push_str}
+}}
+"""
+
+
+# Generates RegisterCodegenUnboxedKernels.cpp.
+@dataclass(frozen=True)
+class ComputeCodegenUnboxedKernels:
+    @method_with_native_function
+    def __call__(self, f: NativeFunction) -> str:
+        # We unconditionally generate function wrappers,
+        sig_group = CppSignatureGroup.from_native_function(
+            f, method=(Variant.method in f.variants)
+        )
+
+        sig = sig_group.most_faithful_signature()
+
+        # escape double quote in schema, get rid of extra double quotes
+        schema = json.dumps(sig.func.__str__())[1:-1]
```

Comment:
Nah, don't do this. Even if it works there is no good reason to use the JSON standard to serialize C++ strings. tools.codegen.gen has an appropriate function to use

```
def cpp_string(s: str) -> str:
    """Convert a python string into a c++ string literal """
    s = s.replace('\\', '\\\\')
    s = s.replace('"', '\\"')
    s = s.replace('\a', '\\a')
    s = s.replace('\b', '\\b')
    s = s.replace('\f', '\\f')
    s = s.replace('\n', '\\n')     
    s = s.replace('\v', '\\v')     
    s = s.replace('\t', '\\t')     
    return f'"{s}"'
```